ComputeExamples.md

```diff
@@ -0,0 +1,701 @@
+      }
+    }
+  },
+  services: [
+    {
+      id: 'notAnId',
+      type: 'access',
+      files: '',
+      datatokenAddress: '0xa15024b732A8f2146423D14209eFd074e61964F3',
+      serviceEndpoint: 'https://v4.provider.goerli.oceanprotocol.com',
+      timeout: 300
+    }
+  ]
+}
+```
+
+Now we define the variables which we will need later
+```Typescript
+let web3: Web3
+let config: Config
+let aquarius: Aquarius
+let datatoken: Datatoken
+let providerUrl: string
+let publisherAccount: string
+let consumerAccount: string
+let addresses
+let computeEnvs
+
+let datasetId: string
+let algorithmId: string
+let resolvedDatasetDdo: DDO
+let resolvedAlgorithmDdo: DDO
+
+let computeJobId: string
+```
+
+### 4.3 Helper methods
+
+Now we define the helper methods which we will use later to publish the dataset and algorithm, and also order them
+
+Add a `createAsset()`function.
+```Typescript
+async function createAsset(
+  name: string,
+  symbol: string,
+  owner: string,
+  assetUrl: Files,
+  ddo: DDO,
+  providerUrl: string
+) {
+  const nft = new Nft(web3)
+  const Factory = new NftFactory(addresses.ERC721Factory, web3)
+
+  // Now we update the DDO and set the right did
+  const chain = await web3.eth.getChainId()
+  ddo.chainId = parseInt(chain.toString(10))
+  const nftParamsAsset: NftCreateData = {
+    name,
+    symbol,
+    templateIndex: 1,
+    tokenURI: 'aaa',
+    transferable: true,
+    owner
+  }
+  const datatokenParams: DatatokenCreateParams = {
+    templateIndex: 1,
+    cap: '100000',
+    feeAmount: '0',
+    paymentCollector: ZERO_ADDRESS,
+    feeToken: ZERO_ADDRESS,
+    minter: owner,
+    mpFeeAddress: ZERO_ADDRESS
+  }
+  // Now we can make the contract call createNftWithDatatoken
+  const result = await Factory.createNftWithDatatoken(
+    owner,
+    nftParamsAsset,
+    datatokenParams
+  )
+
+  const nftAddress = result.events.NFTCreated.returnValues[0]
+  const datatokenAddressAsset = result.events.TokenCreated.returnValues[0]
+  ddo.nftAddress = web3.utils.toChecksumAddress(nftAddress)
+
+  // Next we encrypt the file or files using Ocean Provider. The provider is an off chain proxy built specifically for this task
+  assetUrl.datatokenAddress = datatokenAddressAsset
+  assetUrl.nftAddress = ddo.nftAddress
+  let providerResponse = await ProviderInstance.encrypt(assetUrl, providerUrl)
+  ddo.services[0].files = await providerResponse
+  ddo.services[0].datatokenAddress = datatokenAddressAsset
+  ddo.services[0].serviceEndpoint = providerUrl
+
+  // Next we update ddo and set the right did
+  ddo.nftAddress = web3.utils.toChecksumAddress(nftAddress)
+  ddo.id =
+    'did:op:' + SHA256(web3.utils.toChecksumAddress(nftAddress) + chain.toString(10))
+  providerResponse = await ProviderInstance.encrypt(ddo, providerUrl)
+  const encryptedResponse = await providerResponse
+  const validateResult = await aquarius.validate(ddo)
+```
+<!--
+  assert(validateResult.valid, 'Could not validate metadata')
```

Comment:
The markdown would read better if we could move this test to after the call `await nft.setMetadata` as then the whol eof the `createAsset()` function would be shown in one code block.

Reply:
agreed